### PR TITLE
Update versions for jax/jaxlib release

### PR DESCRIPTION
**Pull request by sharadmv**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Remember to align the itemized text with the first line of an item within a list
 PLEASE REMEMBER TO CHANGE THE '..main' WITH AN ACTUAL TAG in GITHUB LINK.
 -->
 
-## jax 0.3.14 (Unreleased)
+## jax 0.3.14 (June 21, 2022)
```
Comment (skye) [inline CHANGELOG.md:11]: Add the unreleased versions too, so people can easily add new CHANGELOG items

File: setup.py [hunk 2]
```diff
@@ -14,14 +14,14 @@
 
 from setuptools import setup, find_packages
 
-_current_jaxlib_version = '0.3.10'
+_current_jaxlib_version = '0.3.11'
 # The following should be updated with each new jaxlib release.
-_latest_jaxlib_version_on_pypi = '0.3.10'
+_latest_jaxlib_version_on_pypi = '0.3.11'
```
Comment (skye) [inline setup.py:19]: I think this one isn't supposed to be updated until after the new release is pushed

**Review by skye — APPROVED** [review]
LGTM after comments are addressed

File: setup.py (hunk 2 above)
Comment (yashk2810) [inline setup.py:19]: This should be updated after the release as the comment says.

File: setup.py [hunk 3]
```diff
@@ -14,14 +14,14 @@
 
 from setuptools import setup, find_packages
 
-_current_jaxlib_version = '0.3.10'
+_current_jaxlib_version = '0.3.11'
```
Comment (yashk2810) [inline setup.py:17]: Should be 0.3.14

File: CHANGELOG.md (hunk 1 above)
Comment (sharadmv) [reply CHANGELOG.md:15]: Done,

**Review by skye — COMMENTED** [review]
Oops and update the jaxlib version number, as @yashk2810 said offline

File: setup.py (hunk 3 above)
Comment (sharadmv) [reply setup.py]: Done.

File: CHANGELOG.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ PLEASE REMEMBER TO CHANGE THE '..main' WITH AN ACTUAL TAG in GITHUB LINK.
     bucket path as input.
   * Added {func}`jax.scipy.stats.gennorm`.
 
-## jaxlib 0.3.11 (Unreleased)
+## jaxlib 0.3.11 (June 21, 2022)
```
[the remark this answers is not in the document]
Comment (sharadmv) [reply CHANGELOG.md]: Done.

**Pull request merged**